packages/ui/cart/drawer/CartDrawer.tsx

```diff
@@ -39,12 +39,23 @@ import { useCartDrawer } from './hooks/useCartDrawer';
 
 export function CartDrawer() {
    const appContext = useAppContext();
-   const { isOpen, onOpen, onClose, onViewCart } = useCartDrawer();
+   const {
+      isOpen,
+      onOpen,
+      onClose,
+      onViewCart,
+      errorMessages,
+      clearErrorMessages,
+   } = useCartDrawer();
    const isMounted = useIsMountedState();
    const cart = useCart();
    const checkout = useCheckout();
    const isCartEmpty = cart.isFetched && !cart.data?.hasItemsInCart;
-
+   const clearAddToCartErrors = () => clearErrorMessages();
+   const [hasErrors, setHasErrors] = React.useState(errorMessages.length > 0);
+   React.useEffect(() => {
+      setHasErrors(errorMessages.length > 0);
+   }, [errorMessages]);
```

Comment:
Correct me if I'm missing something, but I think we don't need `useState` here since `hasErrors` is derived from `errorMessages`. Synchronizing states in React should be avoided as it frequently results in weird sync bugs https://kentcdodds.com/blog/dont-sync-state-derive-it

```suggestion
   const hasErrors = errorMessages.length > 0;
```